src/weights.jl

```diff
@@ -2,43 +2,112 @@
 ###### Weight vector #####
 
 if VERSION < v"0.6.0-dev.2123"
-    immutable WeightVec{S<:Real, T<:Real, V<:RealVector} <: RealVector{T}
+    abstract AbstractWeights{S<:Real, T<:Real, V<:RealVector} <: RealVector{T}
```

Comment:
Should be `@compat abstract type ... end`. Also why is this being conditioned on the version? Looks like that predates this PR, but I don't see why we would need to condition for these definitions.